amd_openvx_extensions/amd_nn/src/local_response_normalization_layer.cpp

```diff
@@ -0,0 +1,248 @@
+    ERROR_CHECK_STATUS(vxCopyScalar((vx_scalar)parameters[3], &alpha, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    ERROR_CHECK_STATUS(vxCopyScalar((vx_scalar)parameters[4], &beta, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    if(parameters[6]){
+        ERROR_CHECK_STATUS(vxCopyScalar((vx_scalar)parameters[6], &bias, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    }
+    data->mode = miopenLRNCrossChannel;
+    if (type == VX_NN_NORMALIZATION_SAME_MAP) {
+        data->mode = miopenLRNWithinChannel;
+    }
+    else if (type == VX_NN_NORMALIZATION_ACROSS_MAPS) {
+        data->mode = miopenLRNCrossChannel;
+    }
+
+    data->normAlpha = alpha;
+    data->normBeta  = beta;
+    data->normBias  = bias;
+
+    //Input and Output descriptors.
+    ERROR_CHECK_MIOPEN_STATUS((miopenCreateTensorDescriptor(&data->input_desc)));
+    ERROR_CHECK_MIOPEN_STATUS((miopenCreateTensorDescriptor(&data->output_desc)));
+    ERROR_CHECK_MIOPEN_STATUS((miopenSet4dTensorDescriptor(data->input_desc, data_type, input_dims[3], input_dims[2], input_dims[1], input_dims[0])));
+    ERROR_CHECK_MIOPEN_STATUS((miopenSet4dTensorDescriptor(data->output_desc, data_type, output_dims[3], output_dims[2], output_dims[1], output_dims[0])));
+
+    //LRN Descriptor.
+    ERROR_CHECK_MIOPEN_STATUS(miopenCreateLRNDescriptor(&data->lrnDesc));
+    ERROR_CHECK_MIOPEN_STATUS(miopenSetLRNDescriptor(data->lrnDesc, data->mode, data->normN, data->normAlpha, data->normBeta, data->normBias));
+
+    //Input and output memory.
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_OPENCL, &data->input_mem, sizeof(data->input_mem)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[5], VX_TENSOR_BUFFER_OPENCL, &data->output_mem, sizeof(data->output_mem)));
+
+#if ENABLE_DEBUG_PRINT_DIMS
+    std::cout << "lrn input " << input_dims[3] << " " << input_dims[2] << " " << input_dims[1] << " " << input_dims[0] << " ";
+    std::cout << "LRN Mode : " << data->mode << std::endl;
+    std::cout << "Alpha " << data->normAlpha << " Beta " << data->normBeta << " N " << data->normN << " K " << data->normBias << std::endl;
+    std::cout << "output " << output_dims[3] << " " << output_dims[2] << " " << output_dims[1] << " " << output_dims[0] << std::endl;
+#endif
+
+    ERROR_CHECK_STATUS(vxSetNodeAttribute(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK uninitializeLocalResponseNormalizationLayer(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+    LocalResponseNormalizationLayerLocalData * data = NULL;
+    ERROR_CHECK_STATUS(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    ERROR_CHECK_MIOPEN_STATUS(miopenDestroyLRNDescriptor(data->lrnDesc));
+    ERROR_CHECK_MIOPEN_STATUS(miopenDestroyTensorDescriptor(data->input_desc));
+    ERROR_CHECK_MIOPEN_STATUS(miopenDestroyTensorDescriptor(data->output_desc));
```

Comment:
Codacy found an issue: [Either the condition 'if(data)' is redundant or there is possible null pointer dereference: data.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5686893)